contracts/eosio.system/src/delegate_bandwidth.cpp

```diff
@@ -576,12 +579,22 @@ namespace eosiosystem {
             // else (remaining amount is less than genesis balance )
             // -> genesis balance is now decreased to new_delegated_to_self_tokens amount
             genesis.balance = std::min<asset>(del_bw_itr->net_weight + del_bw_itr->cpu_weight, genesis.balance);
+            withdrawn_genesis = prev_genesis - genesis.balance;
          }
        });
 
        if( owner_genesis.balance == zero_asset && owner_genesis.unclaimed_balance == zero_asset){
            genesis_tbl.erase(owner_genesis);
        }
+
+       const int64_t delta_time_usec = (gbm_final_time - current_time_point()).count();
```

Comment:
I think we want to protect against the situation where the withdrawal is after the `gbm_final_time`. Maybe something like
```
const auto withdraw_time = std::min(current_time_point(), gbm_final_time);
const int64_t delta_time_usec = (gbm_final_time - withdraw_time).count();
```

Reply:
It's the least I can do

Reply:
Ok, but I propose:
`check(current_time_point() <= gbm_final_time, "error: withdrawing after final GBM date.");`